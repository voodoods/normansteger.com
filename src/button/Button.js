import React, { Component } from 'react'
import StyledButton from './Button.styled'
import PropTypes from 'prop-types'

class Button extends Component {
  static defaultProps = {
    text: 'Button',
  }

  callRoute(route) {
    window.location.href = route
  }

  render() {
    return (
      <StyledButton onClick={() => this.callRoute(this.props.route)} className="Button" primary={this.props.primary}>
        {this.props.text}
      </StyledButton>
    )
  }
}

Button.propTypes = {
  text: PropTypes.string,
  route: PropTypes.string,
  primary: PropTypes.bool,
}

export default Button
