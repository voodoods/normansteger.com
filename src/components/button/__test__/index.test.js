import React from 'react'
import { shallow } from 'enzyme'
import Button from '../'

describe('<Button /> component index', () => {
  it('expects component to match snapshot', () => {
    const wrapper = shallow(<Button />).dive()
    expect(wrapper).toMatchSnapshot()
  })
})
