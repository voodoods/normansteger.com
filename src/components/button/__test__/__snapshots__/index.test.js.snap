// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> component index expects component to match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    callRoute={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": undefined,
      "className": "Button pure-u-lg-2-5 pure-u-sm-1",
      "onClick": [Function],
      "primary": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  display: inline-block;
  padding: 1em 2em;
  margin: 0 10px 10px 0;
  border-radius: 5px;
  background: #333;
  color: ",
          [Function],
          ";
  background: ",
          [Function],
          ";
  border: 1px solid #000;
  text-align: center;
  cursor: pointer;

  &:hover {
    background: ",
          [Function],
          ";
  }
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": undefined,
        "className": "Button pure-u-lg-2-5 pure-u-sm-1",
        "onClick": [Function],
        "primary": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "rules": Array [
            "
  display: inline-block;
  padding: 1em 2em;
  margin: 0 10px 10px 0;
  border-radius: 5px;
  background: #333;
  color: ",
            [Function],
            ";
  background: ",
            [Function],
            ";
  border: 1px solid #000;
  text-align: center;
  cursor: pointer;

  &:hover {
    background: ",
            [Function],
            ";
  }
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
